tfhe/src/core_crypto/commons/ciphertext_modulus.rs

```diff
@@ -102,6 +102,45 @@ impl<'de, Scalar: UnsignedInteger> serde::Deserialize<'de> for CiphertextModulus
     }
 }
 
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub enum CiphertextModulusCreationError {
+    ModulusTooBig,
+    CustomModuli64BitsOrLessOnly,
+}
+
+impl std::fmt::Debug for CiphertextModulusCreationError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let err_str: &str = (*self).into();
+        write!(f, "{err_str}")
+    }
+}
+
+impl From<CiphertextModulusCreationError> for &str {
+    fn from(value: CiphertextModulusCreationError) -> Self {
+        match value {
+            CiphertextModulusCreationError::ModulusTooBig => {
+                "Modulus is bigger than the maximum value of the associated Scalar type"
+            }
+            CiphertextModulusCreationError::CustomModuli64BitsOrLessOnly => {
+                "Non power of 2 moduli are not supported for types wider than u64"
+            }
+        }
+    }
+}
```

Comment:
Could we have had  a `const fn err_message(self) -> &str`, and then 
```
impl CiphertextModulusCreationError {
    pub const fn const_panic(self) -> ! {
        panic(self.err_message())
    }
}
```

Reply:
for sure, I was battling const stuff and did not quite think of the design and was hoping you'd have a better idea !

Reply:
I think I struggled with formatting the &str though, I'll check

Reply:
Ok I think I tried to use the from/fmt dbg impleme before meaning that the panic was rejecting the code, now with the const err msg is works as expected, will push the update after I check the rest